models/quality_measures/intermediate/cqm438_statin_therapy_for_patients_with_cardiovascular_disease/quality_measures__int_cqm438_denominator_criteria2.sql

```diff
@@ -0,0 +1,184 @@
+
+    select
+          code
+        , code_system
+        , concept_name
+    from {{ ref('quality_measures__value_sets') }}
+    where lower(concept_name) in (
+              'ldl cholesterol'
+            , 'familial hypercholesterolemia'
+        )
+
+)
+
+, conditions as (
+
+    select
+          patient_id
+        , claim_id
+        , encounter_id
+        , recorded_date
+        , source_code
+        , source_code_type
+        , normalized_code
+        , normalized_code_type
+    from {{ ref('quality_measures__stg_core__condition')}}
+
+)
+
+, cholesterol_conditions as (
+
+    select
+          conditions.patient_id
+        , conditions.recorded_date as evidence_date
+    from conditions
+    inner join cholesterol_codes
+        on conditions.source_code_type = cholesterol_codes.code_system
+            and conditions.source_code = cholesterol_codes.code
+
+)
+
+, procedures as (
+
+    select
+          patient_id
+        , procedure_date
+        , coalesce (
+              normalized_code_type
+            , case
+                when lower(source_code_type) = 'cpt' then 'hcpcs'
+                when lower(source_code_type) = 'snomed' then 'snomed-ct'
+                else lower(source_code_type)
+              end
+          ) as code_type
+        , coalesce(
+              normalized_code
+            , source_code
+          ) as code
+    from {{ ref('quality_measures__stg_core__procedure') }}
+
+)
+
+, cholesterol_procedures as (
+
+    select
+          procedures.patient_id
+        , procedures.procedure_date as evidence_date
+    from procedures
+         inner join cholesterol_codes
+             on procedures.code = cholesterol_codes.code
+             and procedures.code_type = cholesterol_codes.code_system
+
+)
+
+, labs as (
+    select
+        patient_id
+        , result
+        , result_date
+        , collection_date
+        , source_code_type
+        , source_code
+        , normalized_code_type
+        , normalized_code
+    from {{ ref('quality_measures__stg_core__lab_result')}}
+
+)
+
+, cholesterol_tests_with_result as (
+    select
+      labs.patient_id
+    , labs.result as evidence_value
+    , coalesce(collection_date,result_date) as evidence_date
+    , cholesterol_codes.concept_name
+    , row_number() over(partition by labs.patient_id order by 
+                          labs.result desc
+                        , result_date desc) as rn
+    from labs
+    inner join cholesterol_codes
+      on ( labs.normalized_code = cholesterol_codes.code
+       and labs.normalized_code_type = cholesterol_codes.code_system )
+      or ( labs.source_code = cholesterol_codes.code
+       and labs.source_code_type = cholesterol_codes.code_system )
+        and {{ apply_regex('labs.result', '[+-]?([0-9]*[.])?[0-9]+') }}
+
+)
+
+, cholesterol_labs as (
+
+    select 
+          patient_id
+        , evidence_date
+    from cholesterol_tests_with_result
+    where rn= 1 
+        and evidence_value > 190
```

Comment:
I think `evidence_value` needs to be type-casted into integer before comparing with the numerical

Reply:
Or could we ensure this column is entirely numeric?

Reply:
The regex ensures we have numeric type in the evidence value. This has been extensively tested in our environment.